radix-engine/src/kernel/call_frame.rs

```diff
@@ -424,14 +425,13 @@ impl CallFrame {
                 .get(&node_id)
                 .ok_or(CallFrameError::RENodeNotVisible(node_id))?;
 
-            to.immortal_node_refs
-                .entry(node_id)
-                .and_modify(|e| {
-                    if e.visibility == RENodeVisibilityOrigin::DirectAccess {
-                        e.visibility = ref_data.visibility
-                    }
-                })
-                .or_insert(ref_data.clone());
+            if let Some(existing_ref_data) = to.immortal_node_refs.get_mut(&node_id) {
+                if existing_ref_data.visibility == RENodeVisibilityOrigin::DirectAccess {
+                    existing_ref_data.visibility = ref_data.visibility;
+                }
+            } else {
+                to.immortal_node_refs.insert(node_id, ref_data.clone());
+            }
```

Comment:
_Note from author:_

Here is the other place where I had to replace `entry()` with an equivalent.
Please see if it is correct AND acceptable from a performance PoV.
(as an alternative, we can switch to an `IndexMap`)

Reply:
N/A.
_(I had some difficulties with importing `.entry()` the right way, but today I learned rust and did it)_